### PR TITLE
add geo tests

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst please review

### CODE REVIEW

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: Please update copyright.

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  def test_geo_cities
+    set_description("Test the nearest neighbor search operator for geo search")
+    deploy_app(SearchApp.new.sd(selfdir + "geo.sd").search_dir(selfdir + "search").threads_per_search(1).enable_http_gateway)
+    start
+    i=0
+    Zlib::GzipReader.open(selfdir + 'c500-r.txt.gz').each_line do |line|
+      line.chomp!
+      lld = line.split(':')
+      assert_equal(3, lld.size)
+      lat, lon, txt = lld
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: Consider creating a helper function that returns the lat, lon and txt from a line to reduce duplication and improve readability.

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      doc = Document.new("geo", get_docid(i, "geo")).
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: Consider creating a helper function that feeds a single geo document based on i, lat, lon, txt to reduce duplication and improve readability.

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    filter = qprops[:filter]
+    text = qprops[:text]
+
+    result = "yql=select * from sources * where [{\"targetNumHits\": #{target_num_hits},"
+    result += "\"approximate\": #{approx}," if approx
+    result += "\"label\": \"nns\"}] nearestNeighbor(#{doc_tensor},#{query_tensor})"
+    result += " and filter contains \"#{filter}\"" if filter
+    result += " or text contains \"#{text}\"" if text
+    result += ";&ranking.features.query(#{query_tensor})={{x:0}:#{x_0},{x:1}:#{x_1}}"
+    result += "&ranking.profile=combined" if qprops[:combined]
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: This is not used so should probably be removed.

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    query_tensor = qprops[:query_tensor]
+    doc_tensor = qprops[:doc_tensor]
+    approx = qprops[:approx]
+    filter = qprops[:filter]
+    text = qprops[:text]
+
+    result = "yql=select * from sources * where [{\"targetNumHits\": #{target_num_hits},"
+    result += "\"approximate\": #{approx}," if approx
+    result += "\"label\": \"nns\"}] nearestNeighbor(#{doc_tensor},#{query_tensor})"
+    result += " and filter contains \"#{filter}\"" if filter
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: This is not used so should probably be removed.

File: tests/search/nearest_neighbor/geo_nns.rb [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    result += " or text contains \"#{text}\"" if text
```
Comment (geirst) [inline tests/search/nearest_neighbor/geo_nns.rb]: This is not used so should probably be removed.

**Comment by arnej27959** [comment]
@geirst PTAL

**Pull request merged**